### ISSUE TITLE
Pass an array to the Router using queryParams

### ISSUE THREAD

**Issue by alexanderpisarev** [angular/angular#14796]
**I'm submitting a ...**
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Currently when we need to pass some query parameter with an array of values, we should do something like this:
```
const queryParams = { ids: [1, 2, 3] };
this.router.navigate(['/some-route'], { queryParams: queryParams });
```
And resulting URL is: `/some-route?ids=1%2C2%2C3`

When we will try to get query parameters via `this.route.queryParams.subscribe((queryParams: Params) =>{...})`, queryParams['ids'] will return string '1,2,3', but not an array with desired values.

**Expected behavior**
Router should serialize queryParams with arrays of values to multiple query parameters with the same name: `/some-route?ids=1&ids=2&ids=3`
There is my investigation of this problem: http://stackoverflow.com/a/42505212/7634393

**Minimal reproduction of the problem with instructions**
http://plnkr.co/edit/eKunV4A7mcOrnnadXfdc
Please, click on the button and look to the browser console.

**What is the motivation / use case for changing the behavior?**
We need to realize search with faceting. Some facets should be multi selectable (for example, via checkboxes). All modifications of facets should be reflected in URL to give to user the ability to send this link via email/skype/slack/something else or to create bookmark on it.
Also, some other peoples have the same problem with sending arrays:
http://stackoverflow.com/questions/42110820/angular-2-pass-array-to-router-querystring
http://stackoverflow.com/questions/42412110/angular-2-how-to-pass-an-array-to-the-router-using-queryparams

* **Angular version:** 2.3.1
create_url_tree#tree() doesnt changed in 2.4.0.
So, bug is still there.

* **Browser:** all

* **Language:** TypeScript 2.0

**Comment by DzmitryShylovich** [comment]
pls add a plunkr

**Comment by alexanderpisarev** [comment]
plunkr added.

**Comment by karolmie1** [comment]
Hey! This fix will be included in 4.0.0-rc.4, right? That will take a while ...
Is there a way to use it right now? When i've tried to install npm by commit, I've got

> Couldn't find manifest in "git://github.com/angular/angular.git"."

Do I have to build locally?
